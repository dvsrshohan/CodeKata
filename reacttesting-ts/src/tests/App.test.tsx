import { render, waitFor } from "@testing-library/react";
import { act, Simulate } from "react-dom/test-utils";
import Counter from "../components/counterApp/counterApp";


test('counter componet should render' ,async ()=>{
    const wrapper = render(<Counter/>)

    const headings = await wrapper.findAllByText('Counter App')
    const button = await wrapper.findByTestId('counter-button')

    expect(headings).toBeTruthy()

    act(()=>{
        Simulate.click(button)
    })

    await waitFor(async()=>{
        const button = await wrapper.findByTestId('counter-button')
        expect(button.innerHTML).toBe("count is 1")
    })

    act(()=>{
        Simulate.click(button)
    })

    await waitFor(async()=>{
        const button = await wrapper.findByTestId('counter-button')
        expect(button.innerHTML).toBe("count is 2")
    })

})




describe("Counter App" , ()=>{
    it("should render with 0 value" , async()=>{
        const wrapper = render(<Counter/>)
        const button = await wrapper.findAllByTestId('counter-button')
        expect(button).toMatchSnapshot('Count value 0')
    })
    it("should render with 1 value" , async()=>{
        const wrapper = render(<Counter/>)
        const button = (await wrapper.findAllByTestId('counter-button')).at(0)
        act(()=>{
            Simulate.click(button as HTMLElement)
        })

        waitFor(()=>{
            expect(button).toMatchSnapshot('count value 1')
        })

    })
})