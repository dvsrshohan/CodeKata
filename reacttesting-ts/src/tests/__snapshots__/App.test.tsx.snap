// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter App should render with 0 value: Count value 0 1`] = `
Array [
  <button
    class="btn btn-tomato col-6"
    data-testid="counter-button"
  >
    count is 
    0
  </button>,
]
`;

exports[`Counter App should render with 1 value: count value 1 1`] = `
<button
  class="btn btn-tomato col-6"
  data-testid="counter-button"
>
  count is 
  1
</button>
`;
